src/ngraph/frontend/onnx_import/utils/reduction.hpp

```diff
@@ -57,22 +57,24 @@ namespace ngraph
                 }
             } // namespace  detail
 
-            /// \brief      Create an nGraph version of an ONNX reduction operation.
             ///
-            /// \param[in]  node          The node representing incoming ONNX operation.
+            /// \brief      Create an nGraph version of an ONNX reduction
+            ///             operation.
             ///
-            /// \tparam     OnnxOperator  Class of an nGraph ArithmeticReduction operation
-            ///                           (e.g. Min, Max, SUm, Product).
+            /// \param[in]  node                The node representing incoming ONNX operation.
+            /// \param[in]  ng_input            The input (nGraph) Tensor.
+            /// \param[in]  reduction_function  The reduction function.
+            ///
+            /// \tparam     ReductionFunction   Functor defining arithmetic reduction operation
+            ///                                 (e.g. Min, Max, SUm, Product).
             ///
             /// \return     nGraph node equivalent of the ONNX operation.
             ///
-            template <class OnnxOperator,
-                      typename std::enable_if<std::is_base_of<ngraph::op::util::ArithmeticReduction,
-                                                              OnnxOperator>::value,
-                                              int>::type = 0>
+            template <class ReductionFunction>
```

Comment:
Why are we removing `std::enable_if` here?

Reply:
We're no longer force type safety by using object from specific type hierarchy.